### PR TITLE
fix: filter form elements events

### PR DESCRIPTION
**Pull request by flatstadt**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?
Added allowIn property to include form elements as event sources

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: projects/ngneat/hotkeys/src/lib/hotkeys.service.ts [hunk 1]
```diff
@@ -115,11 +128,17 @@ export class HotkeysService {
   registerHelpModal(openHelpModalFn: () => void, helpShortcut: string = '') {
     this.addShortcut({ keys: helpShortcut || 'shift.?', showInHelpMenu: false, preventDefault: false }).subscribe(e => {
       const skipMenu =
-        /^(input|textarea)$/i.test(document.activeElement.nodeName) || (e.target as HTMLElement).isContentEditable;
+        /^(input|textarea|select)$/i.test(document.activeElement.nodeName) ||
```
Comment (NetanelBasal) [inline projects/ngneat/hotkeys/src/lib/hotkeys.service.ts:131]: Why not reuse this logic?

File: projects/ngneat/hotkeys/src/lib/hotkeys.service.ts [hunk 2]
```diff
@@ -10,6 +10,7 @@ interface Options {
   group: string;
   element: HTMLElement;
   trigger: 'keydown' | 'keyup';
+  allowIn: string[];
```
Comment (NetanelBasal) [inline projects/ngneat/hotkeys/src/lib/hotkeys.service.ts:13]: we can strongly type the array.

File: projects/ngneat/hotkeys/src/lib/hotkeys.service.ts [hunk 3]
```diff
@@ -77,15 +80,25 @@ export class HotkeysService {
 
     return new Observable(observer => {
       const handler = (e: KeyboardEvent) => {
+        const hotkey = this.hotkeys.get(normalizedKeys);
+        const excludedTargets = this.getExcludedTargets(hotkey.allowIn);
+        const excludedTargetsRegex = new RegExp(`^(${excludedTargets.join('|')})$`);
```
Comment (NetanelBasal) [inline projects/ngneat/hotkeys/src/lib/hotkeys.service.ts:85]: There is no need to run this line if the array is empty

Comment (NetanelBasal) [inline projects/ngneat/hotkeys/src/lib/hotkeys.service.ts:85]: Why we need a regex expression?

File: projects/ngneat/hotkeys/src/lib/hotkeys.service.ts (hunk 1 above)
Comment (flatstadt) [reply projects/ngneat/hotkeys/src/lib/hotkeys.service.ts:131]: document.activeElement.nodeName doesn't work when testing. I guess you can type text in an input programatically without being active.

Comment (NetanelBasal) [reply projects/ngneat/hotkeys/src/lib/hotkeys.service.ts:131]: We can mock it on test. Updating the value programatically will not trigger the callback. Check it.

Comment (flatstadt) [reply projects/ngneat/hotkeys/src/lib/hotkeys.service.ts:131]: I'm using spectator's focus, but doesn't seem to change the document active element.

```
    spectator.focus(spectator.element);
    spectator.dispatchKeyboardEvent(spectator.element, 'keydown', 'a', spectator.element);
```

Comment (NetanelBasal) [reply projects/ngneat/hotkeys/src/lib/hotkeys.service.ts:131]: Yes, just do `document.activeElement.nodeName = value`

File: projects/ngneat/hotkeys/src/lib/hotkeys.service.ts [hunk 4]
```diff
@@ -81,13 +82,9 @@ export class HotkeysService {
     return new Observable(observer => {
       const handler = (e: KeyboardEvent) => {
         const hotkey = this.hotkeys.get(normalizedKeys);
-        const excludedTargets = this.getExcludedTargets(hotkey.allowIn);
-        const excludedTargetsRegex = new RegExp(`^(${excludedTargets.join('|')})$`);
-
-        // Added srcElement as a fallback when target is not present, i.e. during testing.
-        const skipShortcutTrigger =
-          excludedTargets && excludedTargetsRegex.test(((e.target || e.srcElement) as HTMLElement).nodeName);
+        const excludedTargets = this.getExcludedTargets(hotkey.allowIn || []);
```
Comment (NetanelBasal) [inline projects/ngneat/hotkeys/src/lib/hotkeys.service.ts:85]: Much better :)

**Comment by NetanelBasal** [comment]
Please update the docs. You can take the text from the repo.

File: README.md [hunk 5]
```diff
@@ -107,6 +107,8 @@ interface Options {
   element: HTMLElement;
   // The type of event (defaults to `keydown`)
   trigger: 'keydown' | 'keyup';
+  // Allow input, textarea and select as key event sources (defaults to []). AllowInElement can be 'INPUT', 'TEXTAREA' or 'SELECT'.
```
Comment (NetanelBasal) [inline README.md:110]: I meant adding this [section](https://github.com/chieffancypants/angular-hotkeys#allowing-hotkeys-in-form-elements) also in our docs :)

**Pull request merged**